Formula/portable-ruby.rb

```diff
@@ -3,36 +3,39 @@
 class PortableRuby < PortableFormula
   desc "Powerful, clean, object-oriented scripting language"
   homepage "https://www.ruby-lang.org/"
-  # This is the version shipped in macOS 11.7.1/12.6.1/13.
-  url "https://cache.ruby-lang.org/pub/ruby/2.6/ruby-2.6.10.tar.xz"
-  sha256 "5fd8ded51321b88fdc9c1b4b0eb1b951d2eddbc293865da0151612c2e814c1f2"
+  url "https://cache.ruby-lang.org/pub/ruby/3.2/ruby-3.2.2.tar.gz"
```

Comment:
Let's go to Ruby 3.1 first so we have a slightly more liberal requirement here.

Reply:
Liberal for systems that need a system Ruby like Linux arm64? Makes sense.

Might be worth figuring out what the update policy here would be. How often do we update and to which versions? (New Rubies are released on Christmas and are supported for 3 years 3 months)

Reply:
I think once we're on 3.x we can just use whatever version is latest since there wouldn't really be a reason for anything else.

Reply:
> Honestly, not really that much anymore. There may be build script changes between minor versions, but patch versions have never really been a problem with the new workflow.

Glad to hear it. I'd still love to figure out if there's a way we can better share the work in Homebrew/homebrew-core here, though.

> So Ruby 3.2 on 1st April 2024 and every year thereafter? That can work.

Sounds about right, yeh.

Reply:
> we can better share the work in Homebrew/homebrew-core here, though.

What specifically do you have in mind that needs sharing? Most of the shared stuff at the moment is actually in Homebrew/brew (`brew pr-upload` etc.).

Reply:
Well, all of the formulae here are essentially slightly modified versions of Homebrew/homebrew-core ones. It'd be nice if we could share more code (and maintain version updates) while having these work the same. Similarly, I think having Homebrew work with the Ruby installed by `brew install ruby` would be desirable, too (although obviously would need bootstrapped by `portable-ruby`).

Reply:
Ruby is definitely not slightly modified - it's quite majorly modified. The `install` script in homebrew-core is 73 lines, of which only 3 of them are in portable-ruby unmodified. Rust in homebrew-core's Ruby is also a problem for here, as that will increase the compile time from ~7 minutes to 3.5-4 hours (also bad if we force `brew install ruby`).

I definitely think we could have automated bumping though, Maybe a workflow that checks the latest versions in homebrew-core (or formulae.brew.sh) and does `brew bump`? If we do that, the only piece of automation missing would be the automatic opening of PRs in Homebrew/brew (will require a PAT) - the rest of the release process is already one click.

On a related note, Ruby 3.3 will allow us to drop `portable-libedit` and `portable-ncurses` I think which is nice.

Reply:
> also bad if we force `brew install ruby`

I agree that we shouldn't require a `brew install ruby`, but it might be nice to use it if it's available, especially if we can bootstrap from macOS' `/usr/bin/ruby` (but that admittedly might be too complicated).

I'm also a little concerned about what `brew upgrade ruby` might do after the automatic `brew cleanup ruby`.

> I definitely think we could have automated bumping though, Maybe a workflow that checks the latest versions in homebrew-core (or formulae.brew.sh) and does `brew bump-formula-pr`?

It should be enough to parse the output of `brew info --json` here, I think.

> On a related note, Ruby 3.3 will allow us to drop `portable-libedit` and `portable-ncurses` I think, which is nice.

:tada

Reply:
> especially if we can bootstrap from macOS' `/usr/bin/ruby` (but that admittedly might be too complicated).

Yeh, don't want to require bootstrapping with EOL Ruby.

> I agree that we shouldn't require a `brew install ruby`, but it might be nice to use it if it's available

Agreed. Performance will likely be better.

> I'm also a little concerned about what `brew upgrade ruby` might do after the automatic `brew cleanup ruby`.

We've handle such things with portable Ruby cleanup in `brew cleanup` already so shouldn't be hard to do same here.